### PR TITLE
API-26966-526-v2-req-body-updates & API-27054

### PR DESCRIPTION
**Pull request by stiehlrod**
## Summary

- Updates 526 Request body's 'Example Value' to align with validation
- Updates enum for /homeless/riskOfBecomingHomeless/livingSituationOptions
- Fixes validation errors (caused by example.json not matching form_526_json_api.json)
- Adds verify access for ticket 27054

## Related issue(s)
- [API-26966](https://vajira.max.gov/browse/API-26966)
- [API-27054](https://jira.devops.va.gov/browse/API-27054)

## Testing done

- rspec
- Postman

## What areas of the site does it impact?
	modified:   modules/claims_api/config/schemas/v2/526.json
	modified:   modules/claims_api/config/schemas/v2/request_bodies/disability_compensation/example.json
	modified:   modules/claims_api/config/schemas/v2/request_bodies/disability_compensation/request.json
	modified:   modules/claims_api/spec/fixtures/v2/veterans/disability_compensation/form_526_json_api.json
	modified:   modules/claims_api/spec/lib/claims_api/v2/disability_compensation_pdf_mapper_spec.rb
	modified:   modules/claims_api/spec/requests/v2/veterans/disability_compensation_request_spec.rb

## Acceptance criteria

- [X]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [X]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

**Comment by FonzMP** [comment]
dev/swagger needs updated with this.

File: modules/claims_api/config/schemas/v2/request_bodies/disability_compensation/example.json [hunk 1]
```diff
@@ -6,55 +6,52 @@
...
+            "internationalTelephone": "5555555555"
+        },
+        "vaFileNumber": "AB123CDEF"
+    },
+    "changeOfAddress": {
+      "dates": {
+          "beginningDate": "2012-11-31",
+          "endingDate": "2013-10-11"
+        },
+        "typeOfAddressChange": "TEMPORARY",
```
Comment (rockwellwindsor-va) [inline modules/claims_api/config/schemas/v2/request_bodies/disability_compensation/example.json:35]: Looking at the Ticket, does"typeOfAddressChange" need to be changed to "PERMANENT"?  (let me know if I misunderstood and I will resolve this)

File: modules/claims_api/config/schemas/v2/request_bodies/disability_compensation/example.json [hunk 2]
```diff
@@ -6,55 +6,52 @@
...
+            "telephone": "5555555555",
+            "internationalTelephone": "5555555555"
         },
-        "riskOfBecomingHomeless": {
-          "livingSituationOptions": "other living situation",
-          "otherDescription": ""
+        "currentlyHomeless": {
+            "homelessSituationOptions": "FLEEING_CURRENT_RESIDENCE",
+            "otherDescription": "ABCDEFGHIJKLM"
         }
```
Comment (rockwellwindsor-va) [inline modules/claims_api/config/schemas/v2/request_bodies/disability_compensation/example.json:53]: It looks like the "riskOfBecomingHomeless" object got removed, or changed to "currentlyHomeless"?  It seemed "riskOfBecomingHomeless" and then inside that, "livingSituationOptions" still needed to be there?

File: modules/claims_api/config/schemas/v2/request_bodies/disability_compensation/example.json (hunk 1 above)
Comment (stiehlrod) [reply modules/claims_api/config/schemas/v2/request_bodies/disability_compensation/example.json:35]: yes, i lost several changes in the last merge conflict :( thanks!

File: modules/claims_api/config/schemas/v2/request_bodies/disability_compensation/example.json (hunk 2 above)
Comment (rockwellwindsor-va) [reply modules/claims_api/config/schemas/v2/request_bodies/disability_compensation/example.json:53]: I'm still not seeing "riskOfBeingHomeless" in the "example" in the swagger docs, but it needs to be there right?
Screenshot 2023-06-05 at 9 09 44 AM

File: modules/claims_api/config/schemas/v2/request_bodies/disability_compensation/example.json (hunk 1 above)
Comment (rockwellwindsor-va) [reply modules/claims_api/config/schemas/v2/request_bodies/disability_compensation/example.json:35]: @stiehlrod   Looks like it is still "TEMPORARY",  am I looking for that AC point in the right file, or is it only for the example file in request.json?

File: modules/claims_api/config/schemas/v2/request_bodies/disability_compensation/example.json (hunk 2 above)
Comment (stiehlrod) [reply modules/claims_api/config/schemas/v2/request_bodies/disability_compensation/example.json:53]: what do you think?

**Review by rockwellwindsor-va — APPROVED** [review]
Looks good to us!

**Pull request merged**